Libraries/LibWebView/SourceHighlighter.cpp

```diff
@@ -18,11 +18,48 @@
 namespace WebView {
 
 SourceDocument::SourceDocument(StringView source)
-    : m_source(source)
 {
-    m_source.for_each_split_view('\n', AK::SplitBehavior::KeepEmpty, [&](auto line) {
-        m_lines.append(Syntax::TextDocumentLine { *this, line });
-    });
+    // HTML, CSS and JS differ slightly on what they consider a newline to be.
+    // In order to make them get along in documents that include a mix of the three, process the source to make the
+    // newlines consistent before doing any highlighting.
+    // We also replace any invalid characters.
+    // NOTE: We assume the source is utf-8 already, is this OK?
```

Comment:
The only callers originally own a `String` so this is indeed OK.

Since we now store `m_source` as a `String`, we could/should pass the original string as a `String` all the way down from callers of `WebView::highlight_source`. Then the optimization/common path avoids a double allocation.